### ISSUE TITLE
Replace gox and ghr with goreleaser

### ISSUE THREAD

**Issue by kishaningithub** [ory/hydra#1107]
[Go Releaser](https://github.com/goreleaser) lets you do release management (generation of binaries, rpms, debs, brew taps, change logs etc) with a single yaml file.
You can also put in [godownloader](https://github.com/goreleaser/godownloader) to generate an install script that can be used for downloading release binaries from github releases. It is made by the same [goreleaser](https://github.com/goreleaser) team

For inspiration you can take a look at the awesome [golangci-lint](https://github.com/golangci/golangci-lint) project

If you are fine with this change i can put in a PR
 
Bottom line: Installation and release management becomes way cooler 😎

**Comment by aeneasr** [comment]
Sounds good, I'll probably do that myself because the build pipeline is a bit tricky to test.
